src/lib.rs

```diff
@@ -3,7 +3,6 @@
 #![deny(unsafe_code)]
 #![warn(missing_docs)]
 #![warn(clippy::missing_docs_in_private_items)]
-#![forbid(unsafe_code)]
```

Comment:
Why was this removed?

Reply:
Sorry! I definitely meant to comment on this but it slipped my mind when I opened the draft PR.

CSS keywords are case insensitive in the ASCII range. A very common pattern in parsing various bits of CSS syntax is to match an input identifier token against a set of known keywords. [`str::eq_ignore_ascii_case`](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) would be correct, but since it works one byte at a time it’s much slower than `str::eq` / `==`. And it repeats work when expecting one of multiple keywords.

So a more efficient way is to first convert the input to ASCII-lower case (if it’s not already), then compare with `==` against literals that are statically known to already be ASCII-lower-case. This requires space to store the converted input, and allocating on the heap would be slow. Fortunately we have a statically-known upper bound for the size of converted input. The conversion does not change the length, so if the input is longer than all expected keywords there cannot be a match. So we can get away with a `[MaybeUninit<u8>; $MAX_SIZE]` buffer on the stack. [`MaybeUninit::uninit_array`](https://doc.rust-lang.org/nightly/core/mem/union.MaybeUninit.html#method.uninit_array) is not stable yet, so creating that buffer requires an `unsafe` block.

`cssparser` provides macros to do all that, and hide the `unsafe` bits behind an API easily audited for soundness. To enable allocating in the "caller"’s stack frame, that API is in the form of a macro. The lint levels that apply should arguably be those of the macro definition site, but in practice it looks like it’s those of the expansion site. The expansion includes a localized `#[allow(unsafe_code)]`, but that can only counteract a larger `#[deny(unsafe_code)]` (which Taffy’s `lib.rs` does also have on line 3), not `forbid`.

Do you think `forbid` bring a lot of value over `deny`?

Reply:
I'm not convinced that forbid is dramatically better than deny, although it is annoying to have a dependency that requires this.

I'm fine to swap it out for this though.

Reply:
I'm not vehemently against relaxing `forbid(unsafe)` to `deny(unsafe)`. However, in this case I think it's pretty easy to fix this on the `cssparser` side. I've submitted https://github.com/servo/rust-cssparser/pull/335 which I believe keeps the `unsafe` usage encapsulated within `cssparser`.

Reply:
Good idea! I’ve reverted this changed with the new cssparser version.